with-react-compiler/package.json

```diff
@@ -0,0 +1,22 @@
+{
+  "scripts": {
+    "start": "expo start",
+    "android": "expo run:android",
+    "ios": "expo run:ios",
+    "web": "expo start --web",
+    "lint": "expo lint"
+  },
+  "dependencies": {
+    "expo": "^51.0.0",
+    "react": "18.2.0",
```

Comment:
I'm curious if react compiler with react 18 instead of react 19 rc was intentional in this example project?
If so, was it because react 19 wouldn't work with RN and Expo?
If not, why not? It'd be fantastic to have react compiler + react 19 support even in experimental stage with Expo! :)

Reply:
compiler can be used experimentally with React 18, but it requires a polyfill that we're shipping in babel-preset-expo.

Reply:
so, if we were to combine `with-canary-react-19` + `with-canary-new-arch` + `with-react-compiler` examples into one, wouldn't that mean React 19 with React Compiler and New RN Arch?

the reason why I'm asking this is that looking at expo/expo#29168, I can see `react-compiler-runtime.js` polyfill which I assume originates from this gist: https://gist.github.com/poteto/37c076bf112a07ba39d0e5f0645fec43

but if we could just use React 19, even experimentally, why couldn't we get rid of R18 polyfill `react-compiler-runtime.js` entirely?

one more thing: it was also sad to see expo/expo@c1a8c4b revert, it's currently not even possible to use R19 + Compiler + RN NewArch all at the same time :( — are you folks working on this actively some time soon? and how can I help, if I can?